include/ppx/knob.h

```diff
@@ -54,16 +54,15 @@ class Knob
     const std::string& GetFlagName() const { return mFlagName; }
     const std::string& GetFlagHelp() const { return mFlagDesc; }
     size_t             GetIndent() const { return mIndent; }
+    const std::string& GetHelpParams() const { return mHelpParams; }
 
     void SetDisplayName(const std::string& displayName) { mDisplayName = displayName; }
     void SetFlagDesc(const std::string& flagDesc) { mFlagDesc = flagDesc; }
     void SetIndent(size_t indent) { mIndent = indent; }
+    void SetHelpParams(const std::string& helpParams) { mHelpParams = helpParams; }
 
     virtual void Draw() = 0;
 
-    // Returns the flag description used in usage message
-    virtual std::string GetFlagHelpText() const = 0;
```

Comment:
Are `GetFlagName`, `GetFlagHelp` and `GetHelpParams` only used by the KnobManager to build help text?
I  think you moved this function out of Knob because you want to align the description?
You could consider making the KnobManager a friend and keeping some of this interface private, if it doesn't need to be public.